source/robot_model/src/Model.cpp

```diff
@@ -89,19 +89,24 @@ std::vector<std::string> Model::resolve_package_paths_in_urdf(std::string& urdf)
 }
 
 void Model::init_model() {
-  std::ifstream file_stream(this->get_urdf_path());
-  if (!file_stream.is_open()) {
-    throw std::runtime_error("Unable to open file: " + this->get_urdf_path());
-  }
-  std::stringstream buffer;
-  buffer << file_stream.rdbuf();
-  auto urdf = buffer.str();
-
-  pinocchio::urdf::buildModelFromXML(urdf, this->robot_model_);
+  std::ifstream file_stream(this->get_urdf());
+  if (file_stream.is_open()) {
```

Comment:
in the previous version, if that was not the case we would throw. Now we try to parse this as an XML. This holds the risk (which was the case before to be fair), that the string content is not a valid XML and we create an empty pinocchio model (hence the todo comment in lines 101-102)

Reply:
Hmm I think your perspective here on a fully breaking API would be interesting. I wonder if this doesn't maybe make the API less clear now. For example, we could just have a urdf argument that we treat first as xml string then as filepath and whatever it is will be returned in `get_urdf`. What do you think

Reply:
Yes. We could just return emtpy string though. But it's a bit more verbose like that